### PR TITLE
Mempool: completely handle fees and transferred balances within "detectWillFeeExceedBalance()"

### PR DESCRIPTION
**Pull request by andreibancioiu**
## Reasoning behind the pull request
- This is an improvement upon https://github.com/multiversx/mx-chain-go/pull/6603.
- We should have no / as little as possible not-executable transactions in the bulk of selected transactions.
  
## Proposed changes
- https://github.com/multiversx/mx-chain-storage-go/pull/59
- Inform selection about the value transferred within a transaction, so that it can stop selecting transactions from a sender that consumes all the balance (no more balance remaining for fees).

## Testing procedure
- Standard testing
- Craft sequences of transactions that also hold not-executable ones (balance consumed by the first of them).

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by AdoAdoAdo — DISMISSED** [review]

File: process/block/preprocess/selectionSession.go [hunk 1]
```diff
@@ -72,6 +98,58 @@ func (session *selectionSession) IsBadlyGuarded(tx data.TransactionHandler) bool
...
+	}
+
+	esdtTransfers, err := session.esdtTransferParser.ParseESDTTransfers(tx.GetSndAddr(), tx.GetRcvAddr(), function, args)
+	if err != nil {
+		return nil
+	}
+
+	accumulatedNativeValue := big.NewInt(0)
+
+	for _, transfer := range esdtTransfers.ESDTTransfers {
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply process/block/preprocess/selectionSession.go:138]: Very good question, will benchmark ASAP :raised_hands:

[the remark this answers is not in the document]
Comment (andreibancioiu) [reply process/block/preprocess/selectionSession.go:141]:
```
// NOTE: 20% is also due to the require() / assert() calls.
// 0.012993s (TestBenchmarkSelectionSession_GetTransferredValue/numTransactions_=_5_000)
// 0.024580s (TestBenchmarkSelectionSession_GetTransferredValue/numTransactions_=_10_000)
// 0.048808s (TestBenchmarkSelectionSession_GetTransferredValue/numTransactions_=_20_000)
```

Thus, if we select ~10k, about 20 milliseconds (a bit much) will be spent in `GetTransferredValue()`. The selection duration is capped, though. There is still room for optimization.

**We can move the call to `GetTransferredValue()` to the moment the transaction is added to the pool (additions are concurrent, also).**

[the remark this answers is not in the document]
Comment (andreibancioiu) [reply process/block/preprocess/selectionSession.go:141]: MX-16211

File: integrationTests/singleShard/block/executingMiniblocks/executingMiniblocks_test.go [hunk 2]
```diff
@@ -82,12 +82,11 @@ func TestShardShouldNotProposeAndExecuteTwoBlocksInSameRound(t *testing.T) {
...
 
 // TestShardShouldProposeBlockContainingInvalidTransactions tests the following scenario:
-// 1. generate 3 move balance transactions: one that can be executed, one that can not be executed but the account has
-//    the balance for the fee and one that is completely invalid (no balance left for it)
+// 1. generate 3 move balance transactions: one that can be executed, one to be processed as invalid, and one that isn't executable (no balance left for fee).
 // 2. proposer will have those 3 transactions in its pools and will propose a block
 // 3. another node will be able to sync the proposed block (and request - receive) the 2 transactions that
 //    will end up in the block (one valid and one invalid)
-// 4. the non-executable transaction will be removed from the proposer's pool
+// 4. the non-executable transaction will not be immediately removed from the proposer's pool. See MX-16200.
```
Comment (andreibancioiu) [inline integrationTests/singleShard/block/executingMiniblocks/executingMiniblocks_test.go:89]: MX-16200

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**